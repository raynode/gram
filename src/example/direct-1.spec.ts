import {
  graphql,
  GraphQLBoolean,
  GraphQLFloat,
  GraphQLString,
  introspectionQuery,
  printSchema,
} from 'graphql'
import { createSchemaBuilder, NodeType, Service } from '..'

describe('testing the example 1', () => {
  it('should build the example code', async () => {
    const builder = createSchemaBuilder()
    let randomNumber: number

    builder.addQuery({
      name: 'random',
      resolver: () => {
        randomNumber = Math.random()
        return randomNumber
      },
      type: GraphQLFloat,
    })

    const schema = builder.build()

    const result = await graphql({
      schema,
      source: `{ random }`,
    })
    expect(printSchema(schema)).toMatchSnapshot()
    expect(result.data.random).toEqual(randomNumber)
  })
})
