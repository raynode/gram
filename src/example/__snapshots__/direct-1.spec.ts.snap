// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the example 1 should build the example code 1`] = `
"scalar DateTime

interface List {
  page: Page
  nodes: [Node!]!
}

interface Node {
  id: ID
  createdAt: DateTime
  updatedAt: DateTime
  deletedAt: DateTime
}

type Page {
  page: Int
  limit: Int
  offset: Int
}

type Query {
  random: Float
}
"
`;
